### DISCUSSION TITLE
Google map API 免費平替＿OpenStreetMap (OSM) API

### DISCUSSION THREAD

**Discussion by JaYanlloz** [1101-GS4719-C/Python-Community-GS4719-C#2204] [優秀Python套件]
不知道有沒有人遇過需要導入地圖API的時候，如果有想必各位第一個想到的會是Google map吧～
但真的開始研究會發現使用起來不僅麻煩，還得花錢。
這次要介紹的就是在普遍應用場景下可以替代google map的API——OpenStreetMap (OSM) API 
這是一個非常實用的工具，特別是對於那些需要使用地圖資料來開發應用程式的開發者來說。
它不僅是開源的，還可以提供地圖查詢、路徑規劃、地理編碼等功能，讓你能夠很輕鬆地取得精確的地理資訊。
我個人覺得 OSM API 最棒的地方就是它的靈活性，無論你是做導航應用還是資料分析，都能夠自由組合不同的功能，滿足各種需求。
比較特別的是每位使用者都可以貢獻資料，協助完善地圖

不過真要說比較麻煩的部分，可能是他只能使用國際地址的方式輸入地址資訊及輸出地址資訊，這與一班台灣使用者的息關略有不同，還得再輸入輸出介面上下點工夫。

**Comment by 112121py** [comment]
感謝版主分享~
我覺得如果要使用的話，可以在處理輸入及輸出的部分使用AI轉換來協助處理，讓使用者還是能以習慣的方式輸入地址，只是轉換上的精確度跟如何實現，這就要需要有實作的人再分享了。
